### PR TITLE
added note on github file size limit

### PR DESCRIPTION
**Pull request by de-code**
This is related to #730 and #1073. I thought while you are considering the perfect option it would be good to update the documentation. It might be good to tell users about the limitations before installing anything.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,12 +22,14 @@ binaries available for Mac, Windows, Linux, and FreeBSD. Check out the
...
 By default, the Git LFS client needs a Git LFS server to sync the large files
 it manages. This works out of the box when using popular git repository
-hosting providers like GitHub, Atlassian, etc. When you host your own
+hosting providers like GitHub*, Atlassian, etc. When you host your own
 vanilla git server, for example, you need to either use a separate
 [Git LFS server instance](https://github.com/git-lfs/git-lfs/wiki/Implementations),
 or use the [custom transfer adapter](docs/custom-transfers.md) with
 a transfer agent in blind mode, without having to use a Git LFS server instance.
 
+Note: The current single file size limit for GitHub is 2GB.
```
Comment (mathstuf) [inline README.md]: I think that either this should have `*` on it (which probably makes markdown parsers confused) or there should be a stronger link between the footnote mark and this note.

Comment (de-code) [reply README.md]: I thought this might come back. I don't mind either way. Which way would you suggest?

Comment (mathstuf) [reply README.md]: There isn't official support for footnotes in markdown. Something easily searchable at least.

Then again, maybe a section on limitations from common providers is warranted? There's also the monthly bandwidth limitation on GitHub at least.

Comment (de-code) [reply README.md]: Moved to a separate _Limitations_ section. I don't know about the other limitations and other people could perhaps add those. Do you think that might be look better now?

**Comment by mathstuf** [comment]
Looks OK to me.

File: README.md [hunk 2]
```diff
@@ -91,6 +91,10 @@ To https://github.com/git-lfs/git-lfs-test
    67fcf6a..47b2002  master -> master
 ```
 
+## Limitations
+
+The current single file size limit for GitHub is 2GB.
```
Comment (larsxschneider) [inline README.md]: I like that. Maybe extend it with `GitHub and GitHub Enterprise`? 

@bturner Do you know if there is a BitBucket/Stash limitation?
@dosire Do you know if there is a the GitLab limitation?

Comment (bturner) [reply README.md]: @larsxschneider I can't speak for Bitbucket Cloud (bitbucket.org), but Bitbucket Server/Stash, at least as of 5.8, does not impose any limits on the size of individual files; it only monitors how much free space is available and ensures users can't completely fill the disk. I can't say that that will never change (there is an open feature request for quotas, for example), but as of 5.8 (the current release) there are no limits.

I'd wager there _are_ limits for Bitbucket Cloud. Perhaps @erikvanzijst knows what they are.

Comment (mathstuf) [reply README.md]: AFAIK, there are no limitations for self-hosted GitLab (CE, probably the same for EE). I don't know about `gitlab.com`.

As for GitHub, there is also a monthly bandwidth limit for free repositories. No idea for private or Enterprise deployments.

**Comment by ttaylorr** [comment]
Hi all -- I think that this is a great thing to document, but I am not sure that the README is an appropriate place for it. My concern is that this section will become a grab-bag of features that Git LFS doesn't currently support, which is something that I think we (a) should have, and (b) shouldn't be the first thing users see when they navigate to this repository.

I'd like the README to be concise and descriptive. I think that one way to do this is to instead fill this section with a link to a wiki page that documents these limitations. I have created one [here](https://github.com/git-lfs/git-lfs/wiki/Limitations), and think that this would be a good direction towards merging this.

Thoughts?

**Comment by de-code** [comment]
Would it make sense to add a link to the Wiki, in particular this section?

My argument would be that Git LFS is specifically designed to work with large files. So if there is a file size limit (in particular for common provider like GitHub), then it would be fair to point potential users prominently to such limitations. I agree that it's good to keep the README concise. That is why I would probably at least scan the whole of it. Just a pointer that there are such limitations in there would be nice.

**Comment by mathstuf** [comment]
I added the quota information (with links) to that wiki page.

**Pull request closed without merging**